File: src/test/java/com/esentri/wrabbitmq/java/TestDomain.java
package com.esentri.wrabbitmq.java;

import com.esentri.wrabbitmq.WrabbitTopic;

public final class TestDomain {

   private TestDomain(){}

   public static final SimpleListenerTopic ListenerTopic1 = new SimpleListenerTopic();
   public static final SimpleReplierTopic ReplierTopic1 = new SimpleReplierTopic();

   public static final class SimpleListenerTopic extends WrabbitTopic {
      private SimpleListenerTopic() {
         super("Test-Topic-1");
      }

      public WrabbitEvent<String> StringEvent = new WrabbitEvent<>(this, "TT1-TE-1");
      public WrabbitEvent<TestObjectObject> TestObjectObjectEvent = new WrabbitEvent<>(this, "TT1-TE-2");
   }

   public static final class SimpleReplierTopic extends WrabbitTopic {
      private SimpleReplierTopic() {
         super("Test-Topic-2");
      }

      public WrabbitEventWithReply<String, Integer> StringToInt = new WrabbitEventWithReply<>(this, "TT2-TE1");
      public WrabbitEventWithReply<TestObjectObject, String> TestObjectObjectToString = new WrabbitEventWithReply<>(this, "TT2-TE2");
   }
}

File: src/test/kotlin/com/esentri/wrabbitmq/TestDomain.kt
package com.esentri.wrabbitmq

object TestDomain {
   object ListenerTopic1: WrabbitTopic("TestTopic-1") {
      val StringEvent = WrabbitEvent<String>(this, "TT1-TE-1")
      val TestObjectObjectEvent = WrabbitEvent<TestObjectObject>(this, "TT1-TE-2")
   }

   object ReplierTopic1: WrabbitTopic("TestTopic-2") {
      val StringToInt = WrabbitEventWithReply<String, Int>(this, "TT2-TE1")
      val TestObjectObjectToString = WrabbitEventWithReply<TestObjectObject, String>(this, "TT2-TE2")
      val TestObjectObjectToTestObjectNumberText = WrabbitEventWithReply<TestObjectObject, TestObjectNumberText>(this, "TT2-TE3")
   }
}
